lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs

```diff
@@ -1841,3 +1861,64 @@ instance HasSeverityAnnotation ClusterLog where
 
 bracketTracer' :: Tracer IO ClusterLog -> Text -> IO a -> IO a
 bracketTracer' tr name = bracketTracer (contramap (MsgBracket name) tr)
+
+-- | Wait until the given epoch starts.
+waitForEpoch
+    :: EpochNo -- ^ Epoch (start) to wait for.
+    -> Text -- ^ Thing we're waiting for (for log message).
+    -> Tracer IO ClusterLog
+    -> TimeInterpreter Identity
+    -> IO ()
+waitForEpoch e thing tr ti = do
+    now <- currentRelativeTime ti
+    let delta = runIdentity $ interpretQuery ti (timeUntilEpoch e now)
+    traceWith tr $ MsgWaitingForEpoch delta e thing
+    threadDelay . fromIntegral . fromEnum $ realToFrac @_ @Micro delta
+
+-- | Return a TimeInterpreter based on the genesis files in the cluster setup
+-- directory.
+--
+-- Assumes era parameters are changed at the shelley fork at epoch 1, but that
+-- they are not changed further.
+--
+-- The purpose is to concentrate the assumions about test cluster slotting to
+-- one place, and allow us existing @TimeInterpreter@ queries with it.
+timeInterpreterFromTestingConfig
+    :: NetworkParameters  -- ^ Genesis parameters
+    -> FilePath -- ^ Config dir
+    -> IO (TimeInterpreter Identity)
+timeInterpreterFromTestingConfig genesisParams dir = do
+    (shelleyGenesis :: ShelleyGenesis (ShelleyEra Shelley.StandardCrypto))
+        <- either fail pure
+        =<< Aeson.eitherDecodeFileStrict
+        (dir </> "bft" </> "shelley-genesis.json")
+
+    let startTime = StartTime $ sgSystemStart shelleyGenesis
+    let el = sgEpochLength shelleyGenesis
+    let k = SecurityParam $ sgSecurityParam shelleyGenesis
+
+    let byronSlotLength =
+            unSlotLength $ getSlotLength $ slottingParameters genesisParams
+    let tFork =
+            fromRational (toRational $ maxRollbacks k) * 10 * byronSlotLength
+
+    let shelleyFork = HF.Bound
+            (RelativeTime tFork)
+            (W.SlotNo $ Cardano.unEpochSize el)
+            (Cardano.EpochNo 1)
+
+    let byronParams = HF.defaultEraParams k (mkSlotLength 0.1)
+    let shelleyParams = HF.defaultEraParams k (mkSlotLength 0.2)
```

Comment:
This assumes that `k` is the same and would break if we change `k`. Since we can easily ready the byron-genesis from here, I'd not make this assumption in this function and simply read `k` and the slot length from both genesis files.

Reply:
Sure, could do that. 

I didn't because I am thinking about turning things around anyway: have the code be the source of truth, and write to the genesis files, which I imagine would be useful for running integration tests on multiple eras.

Reply:
Done. Used the already-parsed genesis-parameters.